tests/dateSelector.spec.ts

```diff
@@ -0,0 +1,169 @@
+    await expect(newPetTable.locator('dd').first()).not.toHaveText('Tom')
+
+})
+
+test('Select the desired date in the calendar - 2', async ({page}) => {
+
+    /* 2. In the list of the Owners, locate the owner by the name "Jean Coleman" and select this owner */
+    await page.getByText('Jean Coleman').click()
+
+    /* 3. In the list of pets, locate the pet with a name "Samantha" and click "Add Visit" button */
+    // Create two new vet visits for the pet Samantha
+    const petRecord = page.locator('app-pet-list', {hasText: 'Samantha'})
+    await petRecord.getByText('Add Visit').click()
+
+    /* 4. Add the assertion that "New Visit" is displayed as header of the page */
+    await expect(page.locator('h2')).toHaveText('New Visit')
+
+    /* 5. Add the assertion that pet name is "Samantha" and owner name is "Jean Coleman" */
+    // Assert that the pet name and owner are correct
+    await expect(page.locator('table').first().locator('td').first()).toHaveText('Samantha')
+    await expect(page.locator('table').first().locator('td').last()).toHaveText('Jean Coleman')
+
+    /* 6. Click on the calendar icon and select the current date in date picker */
+    // Add an appointment for today's date for 'dermatology'
+    await page.locator('.mdc-icon-button').click()
+    await page.locator('.mat-calendar-body-today').click()
+
+    let date = new Date()
+    const appointmentYear1 = date.getFullYear()
+    const appointmentMonth1 = date.toLocaleString('En-US', {month: '2-digit'})
+    const appointmentDay1 = date.toLocaleString('En-US', {day: '2-digit'})
+    const expectedAppointmentDate1 = `${appointmentYear1}-${appointmentMonth1}-${appointmentDay1}`
+
+    /* 7. Add assertion that date is displayed in the format "YYYY/MM/DD" */
+    // Compare the calendar input to the format we're expecting - "YYYY/MM/DD"
+    await expect(page.locator('.mat-datepicker-input')).toHaveValue(`${appointmentYear1}/${appointmentMonth1}/${appointmentDay1}`)
+
+    /* 8. Type the description in the field, for example, "dermatologists visit" and click "Add Visit" button */
+    await page.locator('#description').fill('dermatologists visit')
+    await page.getByText('Add Visit').click()
+    
+    /* 9. Add assertion that date of visit is displayed at the top of the list of visits for "Samantha" pet on the "Owner Information" page and is in the format "YYYY-MM-DD" */
+    // Validate the appointment date is in the format we're expecting
+    await expect(petRecord.locator('app-visit-list tr td').first()).toHaveText(expectedAppointmentDate1)
+
+    /* 10. Add one more visit for "Samantha" pet by clicking "Add Visit" button */
+    // Add a new vet visit 45 days before today for 'massage therapy'
+    await petRecord.getByText('Add Visit').click() 
+
+    /* 11. Click on the calendar icon and select the date which is 45 days back from the current date */
+    await page.locator('.mdc-icon-button').click()
+
+    date = new Date()
```

Comment:
You don't need to re-initialize the date object again. It's created in line 98 and has a current date. You can just reuse it